samcli/lib/providers/sam_function_provider.py

```diff
@@ -505,6 +552,119 @@ def _parse_layer_info(
 
         return layers
 
+    @staticmethod
+    def _locate_layer_from_nested(  # pylint: disable=too-many-return-statements
```

Comment:
Should we add some debugging logs for this function, in case we need to investigate an issue later?

Reply:
Logs added in the new commit.